### PR TITLE
Order Card

### PR DESCRIPTION
**Pull request by mshago**

### CODE REVIEW

**Comment by mshago** [comment]
image

File: mobile/src/components/orderCard/OrderCard.styles.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  primaryText: {
+    color: color.primary,
+  },
+  orderInfoContainer: {
+    width: '50%',
+    justifyContent: 'space-evenly',
+  },
+  restaurantText: {
+    color: color.black,
+    fontWeight: '600',
```
Comment (lasharela) [inline mobile/src/components/orderCard/OrderCard.styles.ts:64]: Use fobt family instead

**Review by lasharela — CHANGES_REQUESTED** [review]

**Comment by Dulaya** [comment]
Suggestions:
 * Change minWidth: 45% in button to flex: 1
 * Change order.items from array to just number. This will help make it easier to understand for whoever will My Order screen. 
 * Add the green little checkmark next to Starbuck name:

         <View style={{flexDirection: 'row' }}>
            <Text style={styles.restaurantText}>{order.restaurantName}</Text>
            <Icon name="CheckMark" width={8} height={8} />
         </View>

image

 * Use the Icon for bullet point instead (Don't do inline style; Also notice bigger gree bullet):
 
         {delivered && (
            <View style={{flexDirection: 'row', alignItems: 'center'}}>
               <Icon containerStyle={{padding:0, marginRight: 5}} name="Bullet" width={8} height={8} fill="#4EE476" />
               <Text style={styles.orderStatusText}>{order.status}</Text>
            </View>
         )}

image

* Some minor style changes.

**Review by Dulaya — APPROVED** [review]

File: mobile/src/components/orderCard/OrderCard.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+export const OrderCard: FC<TOrderCard> = ({order, handleLeftButton, handleRightButton}) => {
+  const {delivered} = order;
+  return (
+    <View style={[styles.container, styles.shadow]}>
+      <View style={styles.rowContainer}>
+        <View style={[styles.imageContainer, styles.shadow]}>
+          <Image source={{uri: order.restaurantImage}} style={styles.image} />
+        </View>
+        <View style={{width: '50%', justifyContent: 'space-evenly'}}>
```
[the remark this answers is not in the document]
Comment (mshago) [reply mobile/src/components/orderCard/OrderCard.tsx]: done

File: mobile/src/components/orderCard/OrderCard.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          <View style={styles.rowDirection}>
+            {delivered && (
+              <Text style={styles.grayText}>{order.date + ' • '}</Text>
+            )}
+            <Text style={styles.grayText}>{order.items.length + ' Items'}</Text>
+          </View>
+          <Text
+            style={{color: color.black, fontWeight: '600', fontSize: size.m}}>
```
[the remark this answers is not in the document]
Comment (mshago) [reply mobile/src/components/orderCard/OrderCard.tsx]: done

File: mobile/src/components/orderCard/OrderCard.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            </Text>
+            <Text style={{color: color.black, fontSize: 14}}>
+              {order.status}
+            </Text>
+          </View>
+        </View>
+      )}
+      <View style={styles.rowContainer}>
+        <Button
+          title={delivered?'Rate':'Cancel'}
```
[the remark this answers is not in the document]
Comment (mshago) [reply mobile/src/components/orderCard/OrderCard.tsx]: done

**Review by Nap1400 — APPROVED** [review]

**Review by Nap1400 — APPROVED** [review]

File: mobile/src/components/button/Button.type.ts [hunk 5]
```diff
@@ -1,10 +1,10 @@
-import  {GestureResponderEvent, StyleProp, ViewStyle} from 'react-native';
+import  {GestureResponderEvent, StyleProp, TextStyle, ViewStyle} from 'react-native';
 
 export type TButtonProps = {
     title?: string;
     isDisabled?: boolean;
     containerStyle?: StyleProp<ViewStyle>;
-    titleStyle?: StyleProp<ViewStyle>;
+    titleStyle?: StyleProp<TextStyle>;
```
Comment (leo-he-1) [inline mobile/src/components/button/Button.type.ts:7]: Thank you for showing me this change.

**Review by leo-he-1 — APPROVED** [review]

**Review by carizzuti — APPROVED** [review]
This all looks good, I don't see any issues.

**Review by aahenry51 — APPROVED** [review]

File: mobile/src/components/orderCard/OrderCard.styles.ts (hunk 1 above)
Comment (mshago) [reply mobile/src/components/orderCard/OrderCard.styles.ts]: done

**Pull request merged**